scriptmodules/system.sh

```diff
@@ -388,19 +388,32 @@ function get_platform() {
                 __platform="armv7-mali"
                 ;;
             *)
-                # jetson nano and tegra x1 can be identified via /sys/firmware/devicetree/base/model
-                local model_path="/sys/firmware/devicetree/base/model"
-                if [[ -f "$model_path" ]]; then
-                    # ignore end null to avoid bash warning
-                    local model=$(tr -d '\0' <$model_path)
-                    case "$model" in
-                        "NVIDIA Jetson Nano Developer Kit")
-                            __platform="jetson-nano"
-                            ;;
-                        icosa|icosa_emmc)
-                            __platform="tegra-x1"
-                            ;;
-                    esac
+                # jetsons can be identified by device tree or soc0/family (depending on the L4T version used)
+                # refer to the nv.sh script in the L4T DTS for a similar implementation
+                if [[ -e "/proc/device-tree/compatible" ]]; then
+                    CHIP="$(tr -d '\0' < /proc/device-tree/compatible)"
+                    if [[ ${CHIP} =~ "tegra186" ]]; then
+                        __platform="tegra-x2"
+                    elif [[ ${CHIP} =~ "tegra210" ]]; then
+                        __platform="tegra-x1"
+                    elif [[ ${CHIP} =~ "tegra194" ]]; then
+                        __platform="xavier"
+                    fi
+                elif [[ -e "/sys/devices/soc0/family" ]]; then
+                    CHIP="$(tr -d '\0' < /sys/devices/soc0/family)"
+                    if [[ ${CHIP} =~ "tegra20" ]]; then
+                        __platform="tegra-2"
+                    elif [[ ${CHIP} =~ "tegra30" ]]; then
+                        __platform="tegra-3"
+                    elif [[ ${CHIP} =~ "tegra114" ]]; then
+                        __platform="tegra-4"
+                    elif [[ ${CHIP} =~ "tegra124" ]]; then
+                        __platform="tegra-k1-32"
+                    elif [[ ${CHIP} =~ "tegra132" ]]; then
+                        __platform="tegra-k1-64"
+                    elif [[ ${CHIP} =~ "tegra210" ]]; then
+                        __platform="tegra-x1"
+                    fi
```

Comment:
Could this be done with a case statement (and above) using wildcards instead.
Also using bash RE is overkill even without imho. Can use `[[ "$var" == *string* ]]` btw which is 
Faster but prefer a case statement with wildcards.

Please use lower case `chip` and make it a local variable.

Reply:
honestly I knew you were going to suggest using a case statement... I was just lazy and this is how nvidia implemented it in their scripts
I'll make the changes